edn/src/types.rs

```diff
@@ -44,12 +43,63 @@ pub enum Value {
     Map(BTreeMap<Value, Value>),
 }
 
-use self::Value::*;
+/// SpannedValue is the same as Value but used in ValueAndSpan
```

Comment:
nit: `SpannedValue` is parallel to `Value` but container types have `ValueAndSpan` children.